mutation_compactor.hh

```diff
@@ -420,8 +420,7 @@ public:
             _last_pos = rtc.position();
         }
         ++_stats.range_tombstones;
-        do_consume(std::move(rtc), consumer, gc_consumer);
-        return stop_iteration::no;
+        return do_consume(std::move(rtc), consumer, gc_consumer);
```

Comment:
Does this fix a bug?

Reply:
I think this can cause a mutation read (read repair) not stop as soon as the memory checker asks it.
I found this because it breaks range-tombstone based limits, but didn't break any existing test.

Reply:
Maybe it wants to be extracted out to be possibly backported. At least move it to a separate patch with an issue, though I don't think it's serious enough for a backport.

Reply:
Extracted into https://github.com/scylladb/scylla/pull/11139